codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -46,6 +53,7 @@ public boolean visit(Http2Stream stream) {
     private final Http2Connection.PropertyKey stateKey;
     private int initialWindowSize = DEFAULT_WINDOW_SIZE;
     private ChannelHandlerContext ctx;
+    private final List<Listener> listeners = new ArrayList<Listener>(4);
```

Comment:
I was originally wondering if we needed a collection at all.  Just wondering how you were using it.  If it is mostly (always) going to be 1 then maybe just support 1 listener?  This way we don't have to worry about looping around listeners every time we write something.  We could always make a listeners that supports delegating events to a collection of listeners if needed.